### DISCUSSION TITLE
Snowflake via SSO or keypair authentication?

### DISCUSSION THREAD

**Discussion by clayheaton** [Budibase/budibase#10914] [Help] [answered]
Hi - I'm evaluating Budibase. We use Snowflake and either can log in via Azure AD SSO or with a service account keypair. For my purposes, I would like to log in with the keypair. 

Is it possible to create a Budibase Snowflake data source that uses a keypair for authentication? If not, are there any plans to add it?

**Comment by melohagan** [comment, marked as answer]
Hey @clayheaton 

Looking at the nodejs driver docs it does appear there is an option to add this: https://docs.snowflake.com/en/developer-guide/node-js/nodejs-driver-authenticate#using-key-pair-authentication-key-pair-rotation

Initially I imagine we would be more likely to support the service account keypair method as OAuth datasources is a larger piece of work that applies to a number of connectors.

I've raised a feature request here: https://github.com/Budibase/budibase/issues/10932

**Reply by clayheaton** [reply]
Thank you! In many corporate environments, SSO is a requirement for end-user access and keypair authentication is the only allowed way to connect apps to data sources. I appreciate you taking the time to consider this because, without it, Budibase is losing access to a lot of potential users.
